internal/template/funcmap.go

```diff
@@ -58,6 +62,23 @@ func has(m map[string]interface{}, key string) bool {
 	return get(m, key) != nil
 }
 
+// hclField returns a hcl marshaled field e.g. `name = "foo"`
+func hclField(m map[string]interface{}, key string, req bool) (string, error) {
```

Comment:
Turns out you can return errors in template funcs and the template engine will handle it properly (and you don't need to handle it in the template itself). Neat.